File: VB/TreeViewCheckBoxVisibility_vb/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc

Namespace TreeViewContextMenu.Controllers
	Public Class HomeController
		Inherits Controller
		Public Function Index() As ActionResult
			Session("disabledList") = String.Empty
			ViewBag.SelectedIndex = 0
			Return View()
		End Function
		<HttpPost> _
		Public Function FormAction(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal rbList As Integer, ByVal tvhSelectedNodes As String, ByVal tvhCurrentState As String) As ActionResult
			ViewBag.SelectedIndex = rbList
			ViewBag.CurrentState = tvhCurrentState
			ViewBag.SelectedNodeNames = tvhSelectedNodes
			Return View("Index")
		End Function
	End Class
End Namespace

File: CS/TreeViewCheckBoxVisibility/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;

namespace TreeViewContextMenu.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            Session["disabledList"] = string.Empty;
            ViewBag.SelectedIndex = 0;
            return View();
        }
        [HttpPost]
        public ActionResult FormAction([ModelBinder(typeof(DevExpressEditorsBinder))] int rbList, string tvhSelectedNodes, string tvhCurrentState) {
            ViewBag.SelectedIndex = rbList;
            ViewBag.CurrentState = tvhCurrentState;
            ViewBag.SelectedNodeNames = tvhSelectedNodes;
            return View("Index");
        }
    }
}
